liblangutil/CharStream.cpp

```diff
@@ -144,3 +144,32 @@ string CharStream::singleLineSnippet(string const& _sourceCode, SourceLocation c
 
 	return cut;
 }
+
+optional<int> CharStream::translateLineColumnToPosition(LineColumn const& _lineColumn) const
+{
+	return translateLineColumnToPosition(m_source, _lineColumn);
+}
+
+optional<int> CharStream::translateLineColumnToPosition(std::string const& _text, LineColumn const& _input)
+{
+	if (_input.line < 0)
+		return nullopt;
+
+	size_t offset = 0;
+	for (int i = 0; i < _input.line; i++)
+	{
+		offset = _text.find('\n', offset);
+		if (offset == _text.npos)
+			return nullopt;
+		offset++; // Skip linefeed.
```

Comment:
Can we actually assume that the file does not have Windows CFLR line endings?

Reply:
For `\r\n` we are fine, because column numbers higher than the number of characters are truncated to the max value at least for LSP.